Tests/UnitTests/Core/TrackFitting/KalmanFitterTests.cpp

```diff
@@ -22,560 +20,483 @@
+using namespace Acts::Test;
 using namespace Acts::UnitLiterals;
 
-namespace Acts {
-namespace Test {
+using StraightPropagator =
+    Acts::Propagator<Acts::StraightLineStepper, Acts::Navigator>;
+using ConstantFieldStepper = Acts::EigenStepper<Acts::ConstantBField>;
+using ConstantFieldPropagator =
+    Acts::Propagator<ConstantFieldStepper, Acts::Navigator>;
 
-// A few initialisations and definitionas
-using Jacobian = BoundMatrix;
-using Covariance = BoundSymMatrix;
-using Resolution = std::pair<BoundIndices, double>;
-using ElementResolution = std::vector<Resolution>;
-using VolumeResolution = std::map<GeometryIdentifier::Value, ElementResolution>;
-using DetectorResolution =
-    std::map<GeometryIdentifier::Value, VolumeResolution>;
+using KalmanUpdater = Acts::GainMatrixUpdater;
+using KalmanSmoother = Acts::GainMatrixSmoother;
+using KalmanFitter =
+    Acts::KalmanFitter<ConstantFieldPropagator, KalmanUpdater, KalmanSmoother>;
 
-std::normal_distribution<double> gauss(0., 1.);
-std::default_random_engine generator(42);
+/// Find outliers using plain distance for testing purposes.
+///
+/// In a real setup, the outlier classification can be much more involved, e.g.
+/// by computing the weighted distance/ local chi2 value. Here, the purpose is
+/// to test that the basic principle works using simplified, synthetic data.
+/// Thus, the simplest possible implementation should do.
+struct TestOutlierFinder {
+  double distanceMax = std::numeric_limits<double>::max();
+
+  /// Classify a measurement as a valid one or an outlier.
+  ///
+  /// @tparam track_state_t Type of the track state
+  /// @param state The track state to classify
+  /// @retval False if the measurement is not an outlier
+  /// @retval True if the measurement is an outlier
+  template <typename track_state_t>
+  bool operator()(const track_state_t& state) const {
+    // can't determine an outlier w/o a measurement or predicted parameters
+    if (not state.hasCalibrated() or not state.hasPredicted()) {
+      return false;
+    }
+    auto residuals = state.calibrated() - state.projector() * state.predicted();
+    auto distance = residuals.norm();
+    return (distanceMax <= distance);
+  }
+};
 
-ActsSymMatrixD<1> cov1D;
-SymMatrix2D cov2D;
+// Construct a straight-line propagator.
+StraightPropagator makeStraightPropagator(
+    std::shared_ptr<const Acts::TrackingGeometry> geo) {
+  Acts::Navigator navigator(std::move(geo));
+  navigator.resolvePassive = false;
+  navigator.resolveMaterial = true;
+  navigator.resolveSensitive = true;
+  Acts::StraightLineStepper stepper;
+  return StraightPropagator(std::move(stepper), std::move(navigator));
+}
 
-// Create a test context
-GeometryContext tgContext = GeometryContext();
-MagneticFieldContext mfContext = MagneticFieldContext();
-CalibrationContext calContext = CalibrationContext();
+// Construct a propagator using a constant magnetic field along z.
+ConstantFieldPropagator makeConstantFieldPropagator(
```

Comment:
Shouldn't we use the same Stepper setup for the simulation and the reco, i.e. both using the `StraightLineStepper`?

Reply:
I thought so, too. But the original tests where not doing that and so I kept it as-is. In both cases we are using zero-field propagation so it should not make a difference. I can understand that you want to create the hits using the straight line stepper to reduce possible errors.

Reply:
I assume that the straight line stepper and the eigen stepper in zero field have exactly the same results, right? Yeah, it's no difference here.

Reply:
They should, yes. Should we leave it like this for now and see afterwards if we could introduce test w/ consistent propagator and maybe some non-zero field?